.github/workflows/ci.yaml

```diff
@@ -6,64 +8,121 @@ on:
 
-  fmt:
-    name: Rustfmt
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@nightly
+      - name: Use Rust stable
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Build and archive tests
+        run: |
+          cargo llvm-cov nextest-archive \
+            --all-features \
+            --workspace \
+            --archive-file nextest-archive.tar.zst
+
+      - name: Upload archive to workflow
+        uses: actions/upload-artifact@v3
         with:
-          components: rustfmt
-      - run: cargo fmt -- --config group_imports=StdExternalCrate,imports_granularity=Module --check
+          name: nextest-archive
+          path: nextest-archive.tar.zst
 
-  clippy:
-    name: Clippy
+  run-tests:
+    name: Run Tests
     runs-on: ubuntu-latest
+    needs: build-test-artifacts
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@stable
+      - name: Checkout PR
+        uses: actions/checkout@v4
+
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Download archive
+        uses: actions/download-artifact@v3
         with:
-          components: clippy
-      - run: cargo clippy --no-deps --all-features --all-targets -- -D warnings -D clippy::expect_used -D clippy::unwrap_used -D clippy::unimplemented
+          name: nextest-archive
+
+      - name: Run tests
+        run: |
+          cargo llvm-cov nextest \
+            --archive-file nextest-archive.tar.zst
 
-  coverage:
+  codecov:
+    name: Code Coverage
     runs-on: ubuntu-latest
-    env:
-      CARGO_TERM_COLOR: always
+    needs: build-test-artifacts
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - name: Install Rust
-        run: rustup update stable
+      - name: Checkout PR
+        uses: actions/checkout@v4
+
       - name: Install cargo-llvm-cov
         uses: taiki-e/install-action@cargo-llvm-cov
-      - name: Generate code coverage
-        run: cargo llvm-cov --all-features --workspace --codecov --output-path codecov.json
+
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Download archive
+        uses: actions/download-artifact@v3
+        with:
+          name: nextest-archive
+
+      - name: Run tests and output coverage
+        shell: bash
+        run: |
+          cargo llvm-cov nextest \
+            --archive-file nextest-archive.tar.zst \
+            --codecov \
+            --output-path codecov.json
+
       - name: Upload coverage to Codecov
```

Comment:
this is an action we've historically struggled with. 
in the stacks-core repo, it would randomly 5xx error on uploads - so we use a wrapper to retry in case it fails. 
https://github.com/Wandalen/wretry.action is what i'm using if you're interested.

that said, if you're not seeing the same 5xx errors - no reason to change it

Reply:
Ok. Taken also note about 5xx errors. If they start to appear we can change that. So far, so good.